app/packs/src/apps/mydb/elements/details/wellplates/propertiesTab/CustomSizeModal.js

```diff
@@ -0,0 +1,130 @@
+import React, { Component } from 'react';
+import {
+  Button, Modal, FormGroup, FormControl, ControlLabel, Col, Row
+} from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import Wellplate from 'src/models/Wellplate';
+
+export default class CustomSizeModal extends Component {
+  static isPositiveInteger(value) {
+    return !Number.isNaN(value) && Number.isInteger(Number(value)) && Number(value) > 0;
+  }
+
+  static propertyIsInvalid(value) {
+    return value > Wellplate.MAX_DIMENSION;
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      width: props.wellplate.width,
+      height: props.wellplate.height
+    };
+  }
+
+  updateDimensionsFromWellplate() {
+    const { wellplate } = this.props;
+    this.setState({
+      width: wellplate.width,
+      height: wellplate.height
+    });
+  }
+
+  applySizeChange() {
+    const { handleClose, wellplate, triggerUIUpdate } = this.props;
+    const { height, width } = this.state;
+    wellplate.changeSize(width, height);
+    triggerUIUpdate();
+    handleClose();
+  }
+
+  updateDimension(dimesion, value) {
+    if (CustomSizeModal.isPositiveInteger(value)) {
+      this.setState({ [dimesion]: value });
+    }
+  }
+
+  renderApplyButton() {
+    const { height, width } = this.state;
+    const disabled = CustomSizeModal.propertyIsInvalid(height) || CustomSizeModal.propertyIsInvalid(width);
+    return (
+      <Button
+        onClick={() => this.applySizeChange()}
+        disabled={disabled}
+      >
+        Apply
+      </Button>
+    );
+  }
+
+  renderProperty(value, label, propertyName) {
+    const invalidStyleClass = CustomSizeModal.propertyIsInvalid(value)
+      ? 'invalid-wellplate-size' : 'size-without-error';
+    const errorMessage = CustomSizeModal.propertyIsInvalid(value)
+      ? <div className="invalid-wellplate-size-text">Size must be smaller than 100.</div>
+      : null;
+    return (
+      <div className="floating-left">
+        <FormGroup>
+          <ControlLabel>{label}</ControlLabel>
+          <FormControl
+            type="text"
+            value={value}
+            className={invalidStyleClass}
+            onChange={(event) => this.updateDimension(propertyName, event.target.value)}
+          />
+          {errorMessage}
+        </FormGroup>
+      </div>
+    );
+  }
+
+  renderSize() {
+    const { height, width } = this.state;
+    return (
+      <FormGroup>
+        <ControlLabel>Size</ControlLabel>
+        <FormControl.Static>{height * width}</FormControl.Static>
+      </FormGroup>
+    );
+  }
+
+  render() {
+    const { showCustomSizeModal, handleClose } = this.props;
+    const { height, width } = this.state;
+    return (
+      <Modal
+        show={showCustomSizeModal}
+        onHide={handleClose}
+        onShow={this.updateDimensionsFromWellplate}
```

Comment:
This leads to the problem, that the `wellplate `object is undefined in `updateDimenesionsFromWellplate`.
I do not fully understand this, but apparently  `() => { this.updateDimensionsFromWellplate(); }` is working. it is probably related to the context, which is bound to the function (see [SO](https://stackoverflow.com/questions/47679673/how-does-event-handlers-with-arrow-functions-achieve-context-binding/47680408#47680408)). We need to revert this change.

Reply:
You're right, what I suggested applies to *functional* components only, see https://react.dev/learn/responding-to-events#adding-event-handlers.